server/routes/nl.py

```diff
@@ -110,12 +110,27 @@ def _remove_places(query, places_str_found: List[str]):
   return ' '.join(query.split())
 
 
-def _get_place_from_dcid(place_dcid: str) -> Place:
-  place_types = dc.property_values([place_dcid], 'typeOf')[place_dcid]
-  place_type = _get_preferred_type(place_types)
-  place_name = dc.property_values([place_dcid], 'name')[place_dcid][0]
+def _get_place_from_dcids(place_dcids: List[str]) -> List[Place]:
+  places = []
+  place_types_dict = dc.property_values(place_dcids, 'typeOf')
+  place_names_dict = dc.property_values(place_dcids, 'name')
+
+  for p_dcid, p_types in place_types_dict.items():
+    p_types = place_types_dict[p_dcid]
```

Comment:
nit: don't need to lookup map again?

Reply:
Good point. But also, upon some testing I realized that the order in which the propert_values API returns the results does not correspond to the input place_dcids. So now I am iterating over the place_dcids first and then checking for existence for both the type and name. The order  is important to prevent the main_place assignment to be somewhat deterministic so that the first place mentioned (and resolved) is the main place.